draft-ietf-quic-multipath.md

```diff
@@ -455,23 +521,65 @@ the server MAY wait for a short, limited time such as one PTO if a path
 probing packet is received on a new path before sending the
 CONNECTION_CLOSE frame.
 
+Note that PATH_ABANDON frame is also used as a signal for the retirement 
+of the associated Path Identifier. When endpoint received PATH_ABANDON frame,
+it SHOULD not use the associated Path Identifier in future packets, it can 
+only use the Path ID in ACK_MP frames for inflight packets or 
+in MP_RETIRE_CONNECTION_ID frames for CID retirement.
+
 ### Refusing a New Path
 
 An endpoint may deny the establishment of a new path initiated by its
 peer during the address validation procedure. According to
 {{QUIC-TRANSPORT}}, the standard way to deny the establishment of a path
 is to not send a PATH_RESPONSE in response to the peer's PATH_CHALLENGE.
 
-### Effect of RETIRE_CONNECTION_ID Frame {#retire-cid-close}
+If endpoint fails to validate a path and consume an available Path Identifier,
+it SHOULD send a PATH_ABANDON frame to retire the associated Path Identifier. 
+
+
+### Allocating, Consuming and Retiring Connection IDs {#consume-retire-cid}
 
-Receiving a RETIRE_CONNECTION_ID frame causes an endpoint to discard
-the resources associated with that Connection ID. Specifically, the endpoint
-should not use the sequence number of the retired Connection ID anymore in
-any control frames, as the peer will not be able to associate those frames to
-a path and will therefore ignore them. This means an endpoint is also not required
-to acknowledge any late packets carrying that Connection ID and, hence,
-it can remove the list of received packets used to send acknowledgements after
-receiving the RETIRE_CONNECTION_ID frame.
+Each endpoints pre-allocate a Path Identifier for each new Connection ID. 
+The Path Identifier 0 indicates the initial path of the connection. 
+Endpoints SHOULD issue at least one unused Connection ID with unused Path Identifier.
```

Comment:
This seems backwards. Connection IDs are issues for paths, not the other way around.
Also, if I allow you to open 1 million paths, you're probably wise to not issue me 1 million CIDs.

Reply:
Record in #325